src/main/java/roomescape/controller/console/ConsoleController.java

```diff
@@ -0,0 +1,51 @@
+package roomescape.controller.console;
+
+import java.util.EnumMap;
+import java.util.Map;
+import org.springframework.boot.CommandLineRunner;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.stereotype.Controller;
+import roomescape.RoomescapeConsoleApplication;
+import roomescape.view.InputView;
+import roomescape.view.OutputView;
+import roomescape.view.command.AdminCommand;
+
+@Controller
+@ConditionalOnBean(RoomescapeConsoleApplication.class)
+public class ConsoleController {
+
+    private final ReservationConsoleController reservationController;
+    private final TimeSlotConsoleController timeSlotController;
+    private final Map<AdminCommand, CommandExecutor> commandExecutors;
+
+    public ConsoleController(ReservationConsoleController reservationController,
+                             TimeSlotConsoleController timeSlotController) {
+        this.reservationController = reservationController;
+        this.timeSlotController = timeSlotController;
+        this.commandExecutors = new EnumMap<>(AdminCommand.class);
+    }
+
+    public void run() {
+        prepareCommandExecutors();
+        AdminCommand command;
+        do {
+            OutputView.printAdminMenu();
+            command = InputView.getAdminCommand();
+            commandExecutors.get(command).execute();
+        } while (!command.isExit());
+    }
+
+    private void prepareCommandExecutors() {
+        commandExecutors.putAll(Map.of(
+                AdminCommand.RESERVATION_MANAGEMENT, reservationController::menu,
+                AdminCommand.TIME_SLOT_MANAGEMENT, timeSlotController::menu,
+                AdminCommand.EXIT, () -> {}
+        ));
+    }
+
+    @Bean
+    public CommandLineRunner executeOnStartUp() {
+        return args -> this.run();
+    }
```

Comment:
`@Bean`을 `Lite mode`로 진행합니다 ([Reference](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/context/annotation/Bean.html)). `@Configuration` 아래에 넣기 위해서는 `Controller`를 위한 의존성 주입이 많이 필요해서, 간단히 클래스 내부에서 빈을 등록했습니다. 

> Bean methods in lite mode will be treated as plain factory methods by the container (similar to factory-method declarations in XML), with scoping and lifecycle callbacks properly applied. The containing class remains unmodified in this case, and there are no unusual constraints for the containing class or the factory methods.

Reply:
의존성 주입이 많이 필요한가요? 아래처럼 하면 한번의 주입으로 끝나지 않나요?

```suggestion
    @Bean
    public CommandLineRunner executeOnStartUp(ConsoleController controller) {
        return args -> controller.run();
    }
```
엔트리 포인트 (시작지점)을 파악할 수 있으면 더 좋을거 같아 별도 Configuration으로 분리하는게 좋다고 생각해요 (콘솔 어플리케이션의 시작점이라는게 잘 드러나도록)
 
지금 코드에선 console이 시작되는데 어디가 기준점인지 한눈에 안보여서요 (ConsoleController가 가능성 높아보이긴 하지만요)

Reply:
뽑아내서 시작점을 알려주는 게 좋겠네요. 뽑는 만큼 `ComponentScan`을 사용해 볼게요!